central/metadata/service/service_impl.go

```diff
@@ -51,9 +51,10 @@ func (s *serviceImpl) AuthFuncOverride(ctx context.Context, fullMethodName strin
 // GetMetadata returns the metadata for Rox.
 func (s *serviceImpl) GetMetadata(ctx context.Context, _ *v1.Empty) (*v1.Metadata, error) {
 	metadata := &v1.Metadata{
-		BuildFlavor:   buildinfo.BuildFlavor,
-		ReleaseBuild:  buildinfo.ReleaseBuild,
-		LicenseStatus: v1.Metadata_VALID,
+		BuildFlavor:      buildinfo.BuildFlavor,
+		ReleaseBuild:     buildinfo.ReleaseBuild,
+		LicenseStatus:    v1.Metadata_VALID,
+		IsManagedService: env.ManagedCentral.BooleanSetting(),
```

Comment:
Be wary of the compatibility tests.  We ran into this a while back when I tried to update this endpoint.  See https://issues.redhat.com/browse/ROX-14188
Fix to sensor was delivered in 3.74, but this will very likely break any sensor prior to 3.74.

Reply:
@dashrews78 thanks for the warning. Perhaps we could consider adding a `v2/metadata` to circumvent this issue ?

Reply:
@dashrews78 how did you work it around the last time?

Reply:
Ok. Found it in https://issues.redhat.com/browse/ROX-13964

https://github.com/stackrox/stackrox/pull/4289/files#diff-6bc5a2a79499407108fb1651d9a15fe23e006c5e6606b0c2f980de66e74653fd

Reply:
From https://redhat-internal.slack.com/archives/C0313JYKH8W/p1683124636779169?thread_ts=1683115780.729269&cid=C0313JYKH8W we know there are at least 4 ACS CS-connected secured clusters. These will be broken.
There are, of course, options to consider like: break them sneakily, or ask to upgrade. That's where a product person input may help.

I, however, become more skeptical that it's a good way to go.

We also need to check if `false` value also breaks for self-managed. I [launched](https://github.com/stackrox/stackrox/pull/5901#issuecomment-1533155034) version compatibility tests. They should reveal it.

Reply:
Given [this](https://github.com/stackrox/stackrox/pull/5901#issuecomment-1534085785), if there's no way to mark the new attribute as skipped by default in JSON form, we'll **have** to leave `/v1/metadata` unchanged. This means either a new API (e.g. `/v2/metadata` as you suggested) or go back to the problem definition and see if it can be solved differently.

Reply:
@msugakov as far as I understand, this problem will occur only for a certain period of time. e.g. until 4.1. Correct me if I'm wrong, but we basically need to keep this endpoint unchanged until roughly 4.1. Say, 3 months.  

Perhaps adding an `/api/v2` is a bit... overkill for this problem. 

Here's a suggestion that would be an "in-between" (if my above assumptions are correct). Could you weigh in on that (also @dashrews78) ? 

1. Introduce new, immediately deprecated, temporary endpoint, e.g. `/v1/metadata2` (ugly I know), so that we can implement the feature without breaking anything. Or even `/v1/metadata_unstable` or something like that. 
2. When we deem that the `/v1/metadata` is safe to augment, we remove `/v1/metadata2` and port back the logic to `/v1/metadata`

Reply:
> this problem will occur only for a certain period of time. e.g. until 4.1. 

No, the problem will appear after Central builds including new attribute in `/v1/metadata` response is
a) deployed in ACS CS, which may be _sooner_ than 4.1 release because we can deploy more frequently;
b) released for self-managed, which is 4.1.

The problem will impact all Sensors that have version <=3.73.z. These won't be able to start.